### DISCUSSION TITLE
cypress config file crashes at the time of initializing component tests

### DISCUSSION THREAD

**Discussion by falderetes** [cypress-io/cypress#28149] [Questions and Help]
Hello everyone, I am trying to set up component tests for a project and I am getting this error after I finish the configuration and it tries to initialize:

> Your configFile threw an error from: cypress.config.js
> We stopped running your tests because your config file crashed.

**Stack trace:**
```
TypeError: Cannot read property 'module' of undefined
    at module.exports (<Project_path>/webpack.config.js:8:7)
    at makeWebpackConfig (/Users/<My_User>/Library/Caches/Cypress/12.4.1/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/webpack-dev-server/dist/makeWebpackConfig.js:92:17)
    at async createWebpackDevServer (/Users/<My_User>/Library/Caches/Cypress/12.4.1/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/webpack-dev-server/dist/createWebpackDevServer.js:10:32)
    at async Function.devServer.create (/Users/<My_User>/Library/Caches/Cypress/12.4.1/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/webpack-dev-server/dist/devServer.js:94:34)
    at async /Users/<My_User>/Library/Caches/Cypress/12.4.1/Cypress.app/Contents/Resources/app/packages/server/node_modules/@cypress/webpack-dev-server/dist/devServer.js:25:24
![Screenshot 2023-10-26 at 10 12 57](https://github.com/cypress-io/cypress/assets/88545366/30b9ff9f-2837-4a6f-9c19-ebf2e49428f0)
```
![Screenshot 2023-10-26 at 10 12 57](https://github.com/cypress-io/cypress/assets/88545366/c50095cc-f95d-4035-a37f-2bd4149ce657)

**Environment Info:**

- Framework: React
- Bundler: Webpack
- Cypress version: 13.3.3
- Node version: 14.18.0
-  React version: 18.2.0
- OS: macOS Sonoma 14.0 (Apple M1)

**Comment by portyu9** [comment]
Hi @falderetes 

There are a few concrete things to fix/check here.

First, Cypress 13 dropped support for Node 14. For the Cypress 13.3.3 version in your post, move to at least Node 18.

Second, there is a version mismatch in the stack trace:

```text
.../Caches/Cypress/12.4.1/Cypress.app/...
```

while your environment says Cypress `13.3.3`.

Run:

```bash
node --version
npx cypress version
```

and make sure the npm package and installed Cypress binary agree. If they do not, reinstall/repair the Cypress binary/cache.

After that, the important part of the exception is:

```text
TypeError: Cannot read property 'module' of undefined
at module.exports (<Project_path>/webpack.config.js:8:7)
```

That means the failure is coming from **your Webpack configuration**, not from `cypress.config.js` itself. Cypress Component Testing loads your Webpack configuration while creating its dev server, and something referenced on line 8 is `undefined`.

Check whether `webpack.config.js` exports a function that expects arguments supplied by some other build tool. If so, either make that function safe when Cypress loads it, or pass Cypress an explicit compatible `webpackConfig`.

For current Cypress versions, also note that React Component Testing uses Webpack 5 and current Cypress has substantially newer Node requirements.

The screenshot attachment is no longer accessible, so I cannot identify exactly which expression on `webpack.config.js:8` is undefined, but that line is the next place to inspect after correcting the Node/Cypress version mismatch.

Please mark this answer as accepted if it helped, thank you.
